### PR TITLE
add a stretchable icon example

### PR DESCRIPTION
**Pull request by ansis**
Add an example that shows how stretchable icons can be added using `addImage`. The stretchable icons have multiple stretch zones. A version of the image with the stretch zones coloured differently is also included.

Screen Shot 2019-12-05 at 6 41 25 PM

### CODE REVIEW

**Comment by ansis** [comment]
To run this locally you have to replace the image urls with the local url:

```diff
 var images = {
-    "popup": "https://docs.mapbox.com/mapbox-gl-js/assets/popup.png",
-    "popup-debug": "https://docs.mapbox.com/mapbox-gl-js/assets/popup_debug.png"
+    "popup": "/mapbox-gl-js/assets/assets/popup.png",
+    "popup-debug": "/mapbox-gl-js/assets/assets/popup_debug.png"
 };
```

**Comment by ryanhamley** [comment]
This doesn't look the same when I run it locally

Screen Shot 2019-12-05 at 4 19 53 PM

File: docs/pages/example/add-image-stretchable.html [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v11'
+});
+
+var images = {
+    "popup": "https://docs.mapbox.com/mapbox-gl-js/assets/popup.png",
+    "popup-debug": "https://docs.mapbox.com/mapbox-gl-js/assets/popup_debug.png"
+};
+
+console.log(mapboxgl);
```
Comment (ryanhamley) [inline docs/pages/example/add-image-stretchable.html]: console.log

**Comment by ansis** [comment]
You may need to update your submodule

**Comment by andrewharvey** [comment]
I also couldn't work out how to get test it, even after running `git submodule update`, but apart from that looks good to me!

**Comment by ansis** [comment]
It looks like https://github.com/mapbox/mapbox-gl-js-docs/blob/publisher-production/docs/pages/dist links to the `mapbox-gl-js` directory in the same directory this repo is in. So you may need to checkout the right version and `yarn run build-prod-min` there or something like that

File: docs/pages/example/add-image-stretchable.html (hunk 1 above)
Comment (ansis) [reply docs/pages/example/add-image-stretchable.html]: fixed in f22496687bb59d3aae358ad6e144637ab8136f72

**Pull request merged**